### PR TITLE
[EarlyReturn] Register ChangeOrIfReturnToEarlyReturnRector to early-return set

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

File: packages/node-name-resolver/src/NodeNameResolver.php [hunk 1]
```diff
@@ -115,15 +115,10 @@ public function isName(Node $node, string $name): bool
...
     {
         if ($node instanceof MethodCall || $node instanceof StaticCall) {
-            if ($node->name instanceof MethodCall) {
-                return null;
-            }
-            if ($node->name instanceof StaticCall) {
-                return null;
-            }
-            if ($node->name instanceof Identifier) {
+            if ($this->isCallOrIdentifier($node->name)) {
```
Comment (TomasVotruba) [inline packages/node-name-resolver/src/NodeNameResolver.php:118]: :+1:

File: rules/code-quality/src/Rector/Return_/SimplifyUselessVariableRector.php [hunk 2]
```diff
@@ -79,11 +79,8 @@ public function refactor(Node $node): ?Node
         /** @var AssignOp|Assign $previousNode */
         $previousNode = $previousNode->expr;
         $previousVariableNode = $previousNode->var;
-        // has some comment
-        if ($previousVariableNode->getComments() !== []) {
-            return null;
-        }
-        if ($previousVariableNode->getDocComment() !== null) {
+
+        if ($this->hasSomeComment($previousVariableNode)) {
```
Comment (TomasVotruba) [inline rules/code-quality/src/Rector/Return_/SimplifyUselessVariableRector.php:83]: :+1:

File: rules/defluent/src/NodeAnalyzer/FluentChainMethodCallNodeAnalyzer.php [hunk 3]
```diff
@@ -88,6 +85,15 @@ public function isFluentClassMethodOfMethodCall(MethodCall $methodCall): bool
         return true;
     }
 
+    private function isCall(Expr $expr): bool
+    {
+        if ($expr instanceof FuncCall) {
```
Comment (TomasVotruba) [inline rules/defluent/src/NodeAnalyzer/FluentChainMethodCallNodeAnalyzer.php:90]: There is actually `MethodCall` above, not `FuncCall`

File: rules/order/src/PropertyRanker.php [hunk 4]
```diff
@@ -72,4 +66,30 @@ public function rank(Property $property): int
...
+            return true;
+        }
+
+        if ($type instanceof IntegerType) {
+            return true;
+        }
+
+        if ($type instanceof BooleanType) {
+            return true;
+        }
```
Comment (TomasVotruba) [inline rules/order/src/PropertyRanker.php:82]: This should be refactored into array map constant, e.g.:

```php
private const TYPE_TO_RANK = [
    StringType::class => 5,
    // ...
    ArrayType::class => 10,
];
```

Then foreached.

File: rules/defluent/src/NodeAnalyzer/FluentChainMethodCallNodeAnalyzer.php (hunk 3 above)
Comment (samsonasik) [reply rules/defluent/src/NodeAnalyzer/FluentChainMethodCallNodeAnalyzer.php]: fixed.

**Comment by TomasVotruba** [comment]
It seems lots of cases are `instanceof` on multiple types.

It might be word having helper static class, e.g.:

```php
StaticInstanceOf::isOneOf($node, [MethodCall::class, StaticCall::class]
```

What do you think?

File: rules/order/src/PropertyRanker.php (hunk 4 above)
Comment (samsonasik) [reply rules/order/src/PropertyRanker.php]: implemented.

**Comment by samsonasik** [comment]
Yes, probably create `Rector\Core\Util\StaticInstanceOf` ?

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by TomasVotruba** [comment]
Thank you :+1:

**Pull request merged**